[Text Fragment article] Add info about link generation and analytics, remove OT info (#5228)
GoogleChrome/web.dev: src/site/content/en/blog/text-fragments/index.md

authors:
  - thomassteiner
date: 2020-06-17

> updated: 2021-05-03

Author: ```suggestion
updated: 2021-05-10
```

the [extension](https://github.com/GoogleChromeLabs/link-to-text-fragment), for browsers that do not
provide built-in support for Text Fragments where the functionality is implemented in JavaScript.

### Programmatic Text Fragment link generation

> The [polyfill](https://github.com/GoogleChromeLabs/text-fragments-polyfill) contains a file
> `fragment-generation-utils.js` that you can import and generate Text Fragment links with as

Reviewer: ```suggestion
The [polyfill](https://github.com/GoogleChromeLabs/text-fragments-polyfill) contains a file
`fragment-generation-utils.js` that you can import and use to generate Text Fragment links. This is
```

Reviewer: The phrase was grammatically correct, but it sounded awkward to me.

outlined in the code sample below:

```js
const { generateFragment } = await import('https://unpkg.com/text-fragments-polyfill/dist/fragment-generation-utils.js');
const result = generateFragment(window.getSelection());
if (result.status === 0) {
  let url = `${location.origin}${location.pathname}${location.search}`;
  const fragment = result.fragment;
  const prefix = fragment.prefix ?
    `${encodeURIComponent(fragment.prefix)}-,` :
    '';
  const suffix = fragment.suffix ?
    `,-${encodeURIComponent(fragment.suffix)}` :
    '';
  const textStart = encodeURIComponent(fragment.textStart);
  const textEnd = fragment.textEnd ?
    `,${encodeURIComponent(fragment.textEnd)}` :
    '';
  url += `#:~:text=${prefix}${textStart}${textEnd}${suffix}`;
  console.log(url);
}
```

### Obtaining Text Fragments for analytics purposes

Plenty of sites use the fragment for routing, which is why browsers strip out Text Fragments

> as to not break those pages. There is an

Reviewer: ```suggestion
so as to not break those pages. There is an
```

[acknowledged need](https://github.com/WICG/scroll-to-text-fragment/issues/128)
to expose Text Fragments links to pages, for example, for analytics purposes,
but the proposed solution is not implemented yet.

> As a workaround for now, you can use the one-liner script below to extract

Reviewer: ```suggestion
As a workaround for now, you can use the code below to extract
```

the Chrome security team evaluated the risk of implementing scroll on navigation to be manageable.
Other user agents may decide to show a manual scroll UI element instead.

> For sites that still wish to opt-out, we support a

Reviewer: ```suggestion
For sites that wish to opt-out, Chrome supports a
```

[Document Policy](https://wicg.github.io/document-policy/)

> header value that they can send, so user agents will not process Text Fragment URLs.

Reviewer: ```suggestion
header value that they can send so user agents will not process Text Fragment URLs.
```
